sdk/messaging/azservicebus/receiver.go

```diff
@@ -74,72 +77,81 @@ type ReceiverOptions struct {
 	// SubQueue should be set to connect to the sub queue (ex: dead letter queue)
 	// of the queue or subscription.
 	SubQueue SubQueue
+
+	retryOptions utils.RetryOptions
 }
 
 const defaultLinkRxBuffer = 2048
 
 func applyReceiverOptions(receiver *Receiver, entity *entity, options *ReceiverOptions) error {
+
 	if options == nil {
 		receiver.receiveMode = ReceiveModePeekLock
-		return nil
-	}
+	} else {
+		if err := checkReceiverMode(options.ReceiveMode); err != nil {
+			return err
+		}
 
-	if err := checkReceiverMode(options.ReceiveMode); err != nil {
-		return err
+		receiver.receiveMode = options.ReceiveMode
+
+		if err := entity.SetSubQueue(options.SubQueue); err != nil {
```

Comment:
Why was this moved to this `else` clause?

Reply:
Some of the code that applied receiver options was outside of the call so I merged that into here (which means that we always need to calculate and set the entity path).